core/src/main/kotlin/com/strumenta/kolasu/semantics/ScopeProvider.kt

```diff
@@ -91,6 +91,7 @@ class ScopeProviderConfiguration(
 
 // builder
 
+@Deprecated("The corresponding component in the semantics module should be used instead.")
 fun scopeProvider(init: ScopeProviderConfiguration.() -> Unit) = ScopeProviderConfiguration().apply(init)
```

Comment:
Can't we just remove it?

Reply:
We decided to use annotations to maintain back compatibility. I completely agree with removing these classes, perhaps once all tests will be migrated.

Reply:
I will migrate all tests and remove the old Symbol Resolution API classes.